CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contributing In General
+So, you've come here hoping to make Concourse better?  Great!
+Concourse welcomes pull-request on any one of the many open-source
+pieces we currently have within our
+[organization](https://github.com/concourse).
+
+There are only a few ground rules that we like to see respected for
+your pull-request to be considered:
+
+- Make sure that you submit all pull-requests to the develop branch on
+Concourse (they won't build otherwise).
+- Each corresponding change should have an associated test to go along
+with it.  If you are having trouble testing, open an issue to discuss
+it.
+- Please don't forget to update the Concourse 
+[documentation](https://github.com/concourse/concourse/tree/develop/docs) 
+if you make a change to any the behavior (especially in fly).
+- Double check that all of the tests you have written pass on the
+pull-request
+
+With those ground rules out of the way, let's get you setup to work on
+the project!
+
+# Setup
+
+### Tools you will need
+- ruby
+    - use system provided if you have it
+- bosh_cli (ruby-gem and all around bosh goodness)
+    - ```gem install bosh_cli bosh_cli_plugin_micro --no-ri --no-rdoc```
+- direnv (homebrew installed)
+     - ```brew install direnv```
+- fly (grab this from Concourse.ci)
+    - [fly-binary-darwin](https://ci.concourse.ci/api/v1/cli?arch=amd64&platform=darwin)
+    - [fly-binary-linux](https://ci.concourse.ci/api/v1/cli?arch=amd64&platform=linux)
+- go (you can install via homebrew)
+    - ```brew install go```
+- ginkgo (testing framework for go, assuming you grab go first)
+    - ```go get github.com/onsi/ginkgo/ginkgo```
+- postgresql (you can also install this via homebrew)
+    - ```brew install postgresql```
+- virtualbox
+    - [grab it here](https://www.virtualbox.org/wiki/Downloads)
+- vagrant
+    - [grab the latest version](https://www.vagrantup.com/downloads.html)
```

Comment:
Great call @robdimsdale - I wondered how much technical knowledge was to be assumed when writing this.  Should we just brew install as much as we can and leave the rest as an exercise to the reader if they want to configure another way?